configure.ac

```diff
@@ -1,53 +1,96 @@
-                                    [echo "PAPI library not found: HAVE_PAPI=no"])
-
-AC_CHECK_FUNC(backtrace_symbols,[echo "backtrace_symbols found: OK to set HAVE_BACKTRACE=yes"],
-				[echo "backtrace_symbols NOT found: HAVE_BACKTRACE=no"])
-
-unset usempich;
-unset usempi;
-AC_CHECK_LIB(mpich,MPI_Init,[echo "libmpich.a found: OK to set HAVE_MPI=yes";usempich=yes],
-	     AC_CHECK_LIB(mpi,MPI_Init,[echo "libmpi.a found: OK to set HAVE_MPI=yes OK";usempi=yes],[echo "MPI library not found: HAVE_MPI=no"]))
-AC_CHECK_FILE(/proc,[echo "/proc found: HAVE_SLASHPROC=yes"],
-		    [echo "/proc not found: HAVE_SLASHPROC=no"])
-AC_CHECK_LIB(pthread,pthread_mutex_init,[echo "pthreads library found: OK to set PTHREADS=yes"],
-					[echo "pthreads library not found: PTHREADS=no"])
-
-if test -n "${usempich}" ; then
-AC_CHECK_LIB(mpich,iargc,[echo "iargc found in libmpich.a: OK to set HAVE_IARGCGETARG=yes"],	                   [echo "iargc not found in libmpich.a: HAVE_IARGCGETARG=no"])
-fi
-
-if test -n "${usempi}" ; then
-AC_CHECK_LIB(mpi,iargc,[echo "iargc found in libmpi.a: OK to set HAVE_IARGCGETARG=yes"],
-		       [echo "iargc not found in libmpi.a: HAVE_IARGCGETARG=no"])
-fi
-
-AC_CHECK_SIZEOF([void *])
-if test "$ac_cv_sizeof_void_p" = 8; then
-  echo "Pointer size = 8 so BIT64=yes";
-elif test "$ac_cv_sizeof_void_p" = 4; then
-  echo "Pointer size = 4 so BIT64=no";
-fi
+#AC_PROG_FC()
+
+dnl dnl This barfs when CC is pgcc and F77 pgf90
+dnl AC_FC_FUNC(z_zz,z_zz)
+dnl case $z_zz in
+dnl     z_zz__)     echo "Fortran name mangling: UNDERSCORING=-DFORTRANDOUBLEUNDERSCORE" ;;
+dnl     z_zz_)      echo "Fortran name mangling: UNDERSCORING=-DFORTRANUNDERSCORE" ;;
+dnl     Z_ZZ)       echo "Fortran name mangling: UNDERSCORING=-DFORTRANCAPS" ;;
+dnl     *)          echo "Fortran name mangling: UNDERSCORING=" ;;
+dnl esac
+
+dnl dnl For proper handling of const and inline
+dnl AC_C_INLINE
+dnl echo "Inlining: -Dinline=$ac_cv_c_inline"
+dnl dnl AC_FUNC_VPRINTF
+dnl AC_CHECK_LIB(papi,PAPI_library_init,[echo "PAPI library found: OK to set HAVE_PAPI=yes"],
+dnl                                     [echo "PAPI library not found: HAVE_PAPI=no"])
+
+dnl AC_CHECK_FUNC(backtrace_symbols,[echo "backtrace_symbols found: OK to set HAVE_BACKTRACE=yes"],
+dnl 				[echo "backtrace_symbols NOT found: HAVE_BACKTRACE=no"])
+
+dnl unset usempich;
+dnl unset usempi;
+dnl AC_CHECK_LIB(mpich,MPI_Init,[echo "libmpich.a found: OK to set HAVE_MPI=yes";usempich=yes],
+dnl 	     AC_CHECK_LIB(mpi,MPI_Init,[echo "libmpi.a found: OK to set HAVE_MPI=yes OK";usempi=yes],[echo "MPI library not found: HAVE_MPI=no"]))
+dnl AC_CHECK_FILE(/proc,[echo "/proc found: HAVE_SLASHPROC=yes"],
+dnl 		    [echo "/proc not found: HAVE_SLASHPROC=no"])
+dnl AC_CHECK_LIB(pthread,pthread_mutex_init,[echo "pthreads library found: OK to set PTHREADS=yes"],
+dnl 					[echo "pthreads library not found: PTHREADS=no"])
+AC_DEFINE([HAVE_SLASHPROC], [1], [some comment])
```

Comment:
@jmrosinski what is HAVE_SLASHPROC? That is, what is a good comment to go with the macro? Right now I have "some comment". ;-)

Reply:
https://en.wikipedia.org/wiki/Procfs

Reply:
HAVE_SLASHPROC=yes means that the OS supports the "/proc" pseudo-filesystem. If available, GPTL uses it to discover and print current memory usage by the invoking application. If not available, it backs off to "getrusage()" which is less reliable because on many Linux systems it's a placebo that just returns zeros.